drivers/adc/adc_npcx.c

```diff
@@ -84,7 +135,8 @@ static void adc_npcx_isr(void *arg)
 	LOG_DBG("%s: status is %04X\n", __func__, status);
 
 	/* Is end of conversion cycle event? ie. Scan conversion is done. */
-	if (IS_BIT_SET(status, NPCX_ADCSTS_EOCCEV)) {
+	if (IS_BIT_SET(status, NPCX_ADCSTS_EOCCEV) &&
+	    IS_BIT_SET(inst->ADCCNF, NPCX_ADCCNF_INTECEN)) {
```

Comment:
Please replace NPCX_ADCCNF_INTECEN with NPCX_ADCCNF_INTECCEN for removing spurious interrupts.